test/AspNetCoreModule.Test/Framework/InitializeTestMachine.cs

```diff
@@ -65,27 +83,135 @@ public class InitializeTestMachine : IDisposable
         {
-            bool result = true;
-            TestUtility.LogInformation("CheckPerquisiteForANCMTest(): Environment.Is64BitOperatingSystem: {0}, Environment.Is64BitProcess {1}", Environment.Is64BitOperatingSystem, Environment.Is64BitProcess);
-            TestUtility.LogInformation("%ANCMTestFlags%: {0}", Environment.ExpandEnvironmentVariables(ANCMTestFlagsEnvironmentVariable));
+            get
+            {
+                if (_MakeCertExeAvailable == null)
+                {
+                    _MakeCertExeAvailable = false;
+                    try
+                    {
+                        string makecertExeFilePath = TestUtility.GetMakeCertPath();
+                        TestUtility.RunCommand(makecertExeFilePath, null, true, true);
+                        TestUtility.LogInformation("Verified makecert.exe is available : " + makecertExeFilePath);
+                        _MakeCertExeAvailable = true;
+                    }
+                    catch
+                    {
+                        _MakeCertExeAvailable = false;
+                    }
+                }
+                return _MakeCertExeAvailable;
+            }
+        }
 
-            if (Environment.Is64BitOperatingSystem && !Environment.Is64BitProcess)
+        public static string TestRootDirectory
+        {
+            get
             {
-                TestUtility.LogInformation("CheckPerquisiteForANCMTest() Failed: ANCM test should be started with x64 process mode on 64 bit machine; if you run this test on Visual Studio, you should set X64 first after selecting 'Test -> Test Settings -> Default Process Architecture' menu");
-                result = false;
+                return Path.Combine(Environment.ExpandEnvironmentVariables("%SystemDrive%") + @"\", "_ANCMTest");
             }
-            return result;
         }
+
+        private static string _globalTestFlags = null;
+        public static string GlobalTestFlags
+        {
+            get
+            {
+                if (_globalTestFlags == null)
+                {
+                    bool isElevated;
+                    WindowsIdentity identity = WindowsIdentity.GetCurrent();
+                    WindowsPrincipal principal = new WindowsPrincipal(identity);
+                    isElevated = principal.IsInRole(WindowsBuiltInRole.Administrator);
+
+                    // check if this test process is started with the Run As Administrator start option
+                    _globalTestFlags = Environment.ExpandEnvironmentVariables(ANCMTestFlagsEnvironmentVariable);
+                    _globalTestFlags = _globalTestFlags.ToLower();
+                    if (_globalTestFlags.Contains(ANCMTestFlags.UseIISExpressContext.ToLower()) && _globalTestFlags.Contains(ANCMTestFlags.UseFullIISContext.ToLower()))
+                    {
+                        throw new ApplicationException(ANCMTestFlagsEnvironmentVariable + " can't be set with both "
```

Comment:
Probably not a good idea to be throwing exceptions from properties.

Reply:
Maybe make this a method?

Reply:
This is fixed.